crates/store/re_entity_db/src/chunk_index.rs

```diff
@@ -0,0 +1,132 @@
+use std::collections::hash_map::Entry;
+
+use ahash::{HashMap, HashSet};
+use itertools::Itertools as _;
+
+use re_chunk::ChunkId;
+use re_chunk_store::ChunkStoreEvent;
+use re_types_core::ChunkIndexMessage;
+
+/// Info about a single chunk that we know ahead of loading it.
+#[derive(Clone, Debug, Default)]
+pub struct ChunkInfo {
+    /// Do we have the whole chunk in memory?
+    pub fully_loaded: bool,
+}
+
+/// A secondary index that keeps track of which chunks have been loaded into memory.
+///
+/// This is currently used to show a progress bar.
+/// TODO(RR-2999): use this for larger-than-RAM.
+#[derive(Default, Debug, Clone)]
```

Comment:
I'm missing a quick overview of how this relates to ChunkIndexMessage and how it behaves in absence (filling in via regular store messages)